src/RegularizedMoments.cc

```diff
@@ -0,0 +1,689 @@
+    Moments firstTruth;
+    firstTruth << 0,
+                  0.0369003690036900,
+                  0.469976517946998,
+                  -0.000272327446521693,
+                  -0.00291142797372289,
+                  0.00709458010990103;
+
+    if (abs(zeroTruth - zeroRes) > tol) {
+        return false;
+    }
+    return approxEqual(firstRes, firstTruth, tol);
+}
+
+bool testBetaX(double tol) {
+    Moments Q, W;
+    std::tie(Q, W) = buildTestMoments();
+    double zeroRes = BetaX::computeValue(Q, W);
+    double bottom = makeBetaGradDivisor(Q, W);
+    Moments firstRes = BetaX::computeGradient(Q, W, bottom);
+    double zeroTruth = 1.11103656491110;
+    Moments firstTruth;
+    firstTruth << 0,
+                  0,
+                  0,
+                  0.310466905407061,
+                  -0.00373831312952513,
+                  0.0000112532002694914;
+
+    if (abs(zeroTruth - zeroRes) > tol) {
+        return false;
+    }
+    return approxEqual(firstRes, firstTruth, tol);
+}
+
+bool testBetaXY(double tol) {
+    Moments Q, W;
+    std::tie(Q, W) = buildTestMoments();
+    double zeroRes = BetaXY::computeValue(Q, W);
+    double bottom = makeBetaGradDivisor(Q, W);
+    Moments firstRes = BetaXY::computeGradient(Q, W, bottom);
+    double zeroTruth = 0.543777255954378;
+    Moments firstTruth;
+    firstTruth << 0,
+                  0,
+                  0,
+                  0.0205607222123882,
+                  0.261745049520270,
+                  -0.00157657335775577;
+
+    if (abs(zeroTruth - zeroRes) > tol) {
+        return false;
+    }
+    return approxEqual(firstRes, firstTruth, tol);
+}
+
+bool testBetaY(double tol) {
+    Moments Q, W;
+    std::tie(Q, W) = buildTestMoments();
+    double zeroRes = BetaY::computeValue(Q, W);
+    double bottom = makeBetaGradDivisor(Q, W);
+    Moments firstRes = BetaY::computeGradient(Q, W, bottom);
+    double zeroTruth = 1.91680644079168;
+    Moments firstTruth;
+    firstTruth << 0,
+                  0,
+                  0,
+                  0.00136163723260849,
+                  0.0346846138706271,
+                  0.220877927421585;
+
+    if (abs(zeroTruth - zeroRes) > tol) {
+        return false;
+    }
+    return approxEqual(firstRes, firstTruth, tol);
+}
+
+
+bool testConstant(double tol) {
+    Moments Q, W;
+    std::tie(Q, W) = buildTestMoments();
+    double zeroRes = Constant::computeValue(Q, W);
+    Moments firstRes = Constant::computeGradient(Q, W);
+    double zeroTruth = 0.0915084542604366/afw::geom::PI;
+    Moments firstTruth;
+    firstTruth << 0,
+                  -0.000675339145833491/afw::geom::PI,
+                  0.00138137552556848/afw::geom::PI,
+                  -0.0118159430257175/afw::geom::PI,
+                  0.00674319680500958/afw::geom::PI,
+                  -0.00689645127785071/afw::geom::PI;
+
+    if (abs(zeroTruth - zeroRes) > tol) {
+        return false;
+    }
+    return approxEqual(firstRes, firstTruth, tol);
+}
```

Comment:
Again, i'd move all these tests out of here. Since they obscure the implementation of the class.

Reply:
discussed above